jack-and-the-beans/buddies-ios
# BUD-29 View Activity Improvements

This PR should finalize almost everything about View Activity. Here is the list of changes:

- Improve constraints around the board. Resolves #66, fixes #65. Should cut off text at the right points instead of putting them under images.
- Report works. You can tap it, send a report message, and it'll save to the database. _Note, you should not be able to see the report button if you own the activity_
- Fix weird flashing on activity when different things happen. This is because I re-ran the animation code every time anything updated. Now it only runs the animation code when the animation state has changed.
- You can now leave an activity (a confirmation dialogue pops up first) if you are _not_ the owner.
- You can now tap a button and confirm deleting an activity if you are the owner, but it doesn't do anything yet. (next pr)
- The segue to view activity is now a nav controller push instead of show modally. Looks a lot better and allows for swipe back.
- You can now remove a user if you are the owner

### Test:
- Join some activities, leave some activities, remove people from activities.
- Note that everything should happen in real-time. If you're looking at the same activity and two devices, changes to anything about the activity (including removing or adding users) should update nearly instantly.
- Try opening an activity with >4 topics. Then scroll the topics. Make sure there is no flash on the screen.
- Navigate to activities and make sure the back button works, as does swiping back.
- Tap the report button and send a report message. See if it shows up in Firestore.
- Make sure you can't see the report button if you own the activity.
- Make sure you can't see the "remove" button if you do NOT own the activity.
- Make sure long user names are cut off properly before the "remove" button and don't push it off-screen
- Make sure long dates/locations are cutoff before the profile pictures in the mini-description.

**ALSO FIXES #71*

### Screenshots of it working better:
<img width="392" alt="screen shot 2019-02-23 at 6 20 47 pm" src="https://user-images.githubusercontent.com/6265975/53308047-1b784100-386c-11e9-86ec-c1ccf435020b.png">
<img width="395" alt="screen shot 2019-02-23 at 7 09 06 pm" src="https://user-images.githubusercontent.com/6265975/53308051-1f0bc800-386c-11e9-8924-1b20335d806d.png">
<img width="342" alt="screen shot 2019-02-23 at 7 17 34 pm" src="https://user-images.githubusercontent.com/6265975/53308052-229f4f00-386c-11e9-882b-7daefc849fea.png">
